enhancements/20250407-cpu-model-features-selection.md

```diff
@@ -0,0 +1,402 @@
+
+If users select the uncommon one, we'll show the warning message.
+
+![](20250407-cpu-model-features-selection/02-warning-message.png)
+
+Besides custom model, we should also provide [`host-passthrough`](https://kubevirt.io/user-guide/compute/virtual_hardware/#cpu-model-special-cases).
+
+![](20250407-cpu-model-features-selection/03-different-model.png)
+
+![](20250407-cpu-model-features-selection/04-custom-model.png)
+
+> These are just examples. The real one will match our Harvester GUI style.
+
+### Implementation Overview
+
+#### Frontend
+
+Frontend needs to create a new tab in VM creation page and provide a dropdown selection menu for selecting models and features. This selection is also available in the VM template page. Then, calculate the common and uncommon CPU models and features across all nodes. 
+
+- If users select the common one, we'll show the option with green mark.
+- If users select the uncommon one, we'll show the option with yellow mark and a warning message.
+
+The data of CPU models and features are from node's labels:
+
+- The model is from `cpu-feature.node.kubevirt.io/{model}` of node's labels. 
+- The feature is from `cpu-feature.node.kubevirt.io/{feature}` of node's labels.
+
+This is node's labels example:
+
+```yaml
+kind: Node
+metadata:
+  name: my-node-1
+  labels:
+    host-model-cpu.node.kubevirt.io/Common-CPU: "true"
+    cpu-model.node.kubevirt.io/Intel-A: "true"
+    cpu-model.node.kubevirt.io/Common-CPU: "true"
+    cpu-model-migration.node.kubevirt.io/Intel-A: "true"
+    cpu-model-migration.node.kubevirt.io/Common-CPU: "true"
+    cpu-feature.node.kubevirt.io/avx2: "true"
+    cpu-feature.node.kubevirt.io/sse4_2: "true"
+```
+
+The VM spec is:
+
+```yaml
+kind: VirtualMachine
+name: my-vm-2
+spec:
+  template:
+    spec:
+      domain:
+      cpu:
+        model: Intel-A
+        features:
+        - name: avx2
+          policy: require
+```
+
+- `cpu.model` is optional. If omitted, the default value is `host-model`.
+- `cpu.features` is optional.
+  - If `cpu.features` is provided, each item in the array must include a `cpu.features[].name` field. 
+  - `cpu.features[].policy` is optional. if omitted, the default value is `require`.  
+    Valid values are: `force`, `require`, `optional`, `disable`, `forbid`.
+
+Since VM spec in `virtualmachinetemplateversion` is same, please use same logic to fill the model and features.
+
+
+Action Items:
+
+- [ ] Create a new tab in VM creation page.
+- [ ] Create a new tab in VM template page.
+- [ ] Provide a dropdown selection menu for selecting models and features.
+- [ ] Calculate common and uncommon the CPU models and features across all nodes.
+
+#### Backend
+
+Backend should reject the unreasonable request from frontend. When users try to migrate a VM, the `findMigratableNodes` action should return available nodes that matched the selected CPU model and features to avoid scheduling failure.
```

Comment:
the cpu model and features are already available as labels

if we convert the cpu feature requirements to labels used for vm scheduling then `findMigratableNodesByVMI` will already handle the scenario of finding nodes based on scheduling requirements